kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_lagrange_multiplier.h

```diff
@@ -0,0 +1,1234 @@
+        }
+        
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief Corresponds to the previews, but the System's matrix is considered already built and only the RHS is built again
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rA The LHS matrix
+     * @param rDx The Unknowns vector
+     * @param rb The RHS vector
+     */
+    void BuildRHSAndSolve(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        ) override
+    {
+        KRATOS_TRY
+
+        // Resize to the proper size
+        const SizeType total_system_size = (BaseType::mOptions.Is(DOUBLE_LAGRANGE_MULTIPLIER)) ? BaseType::mEquationSystemSize + 2 * BaseType::mSlaveIds.size() : BaseType::mEquationSystemSize + BaseType::mSlaveIds.size();
+        if (rDx.size() != total_system_size) {
+            rDx.resize(total_system_size,  false);
+            TSparseSpace::SetToZero(rDx);
+        }
+
+        // Base build and solve
+        BaseType::BuildRHSAndSolve(pScheme, rModelPart, rA, rDx, rb);
+
+        // Update the Lagrange multiplier solution
+        #pragma omp parallel for
+        for (int i = 0; i < static_cast<int>(BaseType::mSlaveIds.size()); ++i) {
+            mLagrangeMultiplierVector[i] += rDx[BaseType::mEquationSystemSize + i];
+        }
+        
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief Function to perform the build of the RHS.
+     * @details The vector could be sized as the total number of dofs or as the number of unrestrained ones
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rb The RHS vector
+     */
+    void BuildRHS(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemVectorType& rb
+        ) override
+    {
+        KRATOS_TRY
+
+        // Resize again the system to the original size
+        if (rb.size() != BaseType::mEquationSystemSize) {
+            rb.resize(BaseType::mEquationSystemSize, false);
+        }
+
+        // Build the base RHS
+        BaseType::BuildRHS(pScheme, rModelPart, rb);
+
+        KRATOS_CATCH("")
+    }
+
+
+    /**
+     * @brief Builds the list of the DofSets involved in the problem by "asking" to each element
+     * and condition its Dofs.
+     * @details The list of dofs is stores insde the BuilderAndSolver as it is closely connected to the
+     * way the matrix and RHS are built
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     */
+    void SetUpDofSet(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart
+        ) override
+    {
+        KRATOS_TRY;
+
+        BaseType::SetUpDofSet(pScheme, rModelPart);
```

Comment:
please remove the methods that only call the baseclass

Reply:
+1